src/main/java/Duke.java

```diff
@@ -1,10 +1,50 @@
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        ui.welcomeMessage();
+        tasks.showCurrentTasks();
+        ui.showLineBreak();
+        boolean isFinished = false;
+        while (!(isFinished)) {
+            try {
+            String inputFromUser = ui.handleInput();
+               Command c = Parser.parse(inputFromUser);
+            if (c == null) {
+                throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-( ☹ OOPS!!!");
+            }
+               tasks.runCommand(c);
+            if (c.getCommand().equals("bye")) {
+                isFinished = true;
+            }
+        } catch (DukeException e) {
+                System.out.println("Invalid command received: " + e.getMessage());
+            } finally {
+                this.storage.save(tasks);
+            }
+        }
+        ui.terminateMessage();
+    }
+
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
```

Comment:
Consider having this be done by duke(ui) instead of main as it is part of the duke. It might make more sense if you output this after duke loads the storage.